build-tools/jvm/src/main/kotlin/aditogether/buildtools/jvm/JvmOptions.kt

```diff
@@ -0,0 +1,9 @@
+package aditogether.buildtools.jvm
+
+/**
+ * Defines the shared configuration for targets compiled with Java (e.g. JVM, Android, KMM).
+ */
+object JvmOptions {
+    const val WARNINGS_AS_ERRORS: Boolean = true
+    const val JAVA_VERSION: Int = 11
```

Comment:
Personally, I found the UPPER_SNAKE a bit retrò and prefer the UpperCamel.

The pattern is not clear in Kotlin, probably the camel has been started to be used to be coherent between enums and sealed classes; the since enums are constants, also some constants became camel.

I would unify everything to the camel. 
What do you think?

Reply:
I personally prefer the upper snake case for constants but that is subjective.

**But**, I think it's valuable to have them in upper snake case for 3 motivations:
1. This is the official code convention for constants ([here](https://kotlinlang.org/docs/coding-conventions.html#property-names)), quoting:
> Names of constants (properties marked with const, or top-level or object val properties with no custom get function that hold deeply immutable data) should use uppercase underscore-separated names

2. JetBrains is using the same pattern internally (e.g. coroutines)
3. In general this is the standard for OSS projects

Reply:
Thanks for the link; I've always been missing that thin line between a constant and a singleton object and I seen a lot of confusion about that within my current team, thus my suggestion to unify the pattern. 

Let's keep this format then!